packages/slate/src/changes/with-schema.js

```diff
@@ -121,12 +104,22 @@ function normalizeNodeAndChildren(change, node, schema) {
  * @return {Node}
  */
 
-function refindNode(change, node) {
+function refindNode(change, node, lastPath) {
   const { value } = change
   const { document } = value
-  return node.object == 'document'
-    ? document
-    : document.getDescendant(node.key)
+  if (node.object === 'document') {
+    return { node: document, nodePath: lastPath }
+  }
+  const tryNode = document.getDescendantAtPath(lastPath)
```

Comment:
Can you add a `PERF: ` comment above this line, explaining why this is a shortcut that speeds things up lots of the time?

Reply:
Yes, I will.